### DISCUSSION TITLE
postgres.h no such file or directory problem

### DISCUSSION THREAD

**Discussion by FLOKITOTO** [osm-search/Nominatim#2620] [Q&A] [answered]
Hey everyone

I hope you are fine, I've followed the documentation "Installation on Ubuntu 20" and I encountered a problem when running the "sudo make install" command.

I would like to know how could I fix this. 

Thanks in advance and Have a good day 
`
flokito@ubuntu:/srv/nominatim/build$ sudo make install
[ 92%] Built target osm2pgsql_lib
[ 97%] Built target osm2pgsql
[100%] Running external makefile /usr/lib/postgresql/14/lib/pgxs/src/makefiles/pgxs.mk
/srv/nominatim/Nominatim-4.0.1/module/Makefile:11: warning: overriding recipe for target 'install'
/usr/lib/postgresql/14/lib/pgxs/src/makefiles/pgxs.mk:245: warning: ignoring old recipe for target 'install'
/srv/nominatim/Nominatim-4.0.1/module/nominatim.c:1:10: fatal error: postgres.h: No such file or directory
    1 | #include "postgres.h"
      |          ^~~~~~~~~~~~
compilation terminated.
make[3]: *** [<builtin>: nominatim.o] Error 1
make[2]: *** [module/CMakeFiles/nominatim_lib.dir/build.make:61: /srv/nominatim/Nominatim-4.0.1/module/dummy] Error 2
make[1]: *** [CMakeFiles/Makefile2:1037: module/CMakeFiles/nominatim_lib.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
`
doc :
[https://nominatim.org/release-docs/latest/appendix/Install-on-Ubuntu-20/](url)

screenshot
[http://image.noelshack.com/fichiers/2022/09/1/1646063569-vmware-g0adjt6pzq.png](url)

**Comment by mtmail** [comment, marked as answer]
`postgres.h` is usually part of the `libpq-dev` package (`apt-get install libpq-dev`).

The instructions are aimed at a fresh install on a Ubuntu 20.4 LTS (long-term-support). It contains installing `postgresql-server-dev-12 postgresql-12-postgis-3`. In your output I see postgresql 14 mentioned. Was postgresql 14 already installed, or do you have two versions of the database installed now?

**Reply by FLOKITOTO** [reply]
Hey Thank you for your answer that was it 👍

**Reply by FLOKITOTO** [reply]
Hey it's me again,

I installed Nominatim on Ubuntu, on a vmware. I installed Vagrant, the installation was done correctly however when I start Vagrant with vagrant up it does not work: 

If you have an idea I am interested 

Thanks in advance

`
-An error occurred while executing the action on the 'ubuntu-apache' machine. Please handle this error then try again:
`
`
-Error while connecting to libvirt: Error making a connection to libvirt URI qemu:///system?no_verify=1&keyfile=/home/flokito/.ssh/id_rsa:
-Call to virConnectOpen failed: Failed to connect socket to '/var/run/libvirt/libvirt-sock': No such file or directory
`


screenshot :

[https://cdn.discordapp.com/attachments/920346266590916658/948174758246232074/vmware_YhYO3IhQxR.png](url)

**Reply by mtmail** [reply]
`vagrant up` will try to start all machine (Ubuntu, centos, ...), try shutting them all down `vagrant halt` and only start one `vagrant up ubuntu`. The error message is something between `vagrant` and the virtualization provider (vmware). I use a different one (virtualbox) but when I encounter similar error messages a reboot of my computer often works.

**Reply by FLOKITOTO** [reply]
Hey thank you for your answer

I did what you said, and I get the same result

`
flokito@ubuntu:/srv/nominatim/Nominatim-4.0.1$ vagrant up ubuntu
-Bringing machine 'ubuntu' up with 'libvirt' provider...
Error while connecting to libvirt: Error making a connection to libvirt URI qemu:///system?no_verify=1&keyfile=/home/flokito/.ssh/id_rsa:
-Call to virConnectOpen failed: Failed to connect socket to '/var/run/libvirt/libvirt-sock': No such file or directory
`

I searched in the indicated directory "/var/run/libvirt/libvirt-sock" and indeed there is nothing.
 
So I installed libvirt but it still doesn't work

`
Error while creating domain: Error saving the server: Call to virDomainDefineXML failed: End of file while reading data: Input/output error
`

How can I fix my problem? 

Thanks in advance

**Reply by mtmail** [reply]
I'm afraid libvirt issues are a question for a vagrant forum. The error means vagrant cannot access the virtual machine at all, unrelated to Nominatim.

**Reply by FLOKITOTO** [reply]
Oh right, I understand better
I thank you for your help
